doc/source/code_style.txt

```diff
@@ -0,0 +1,209 @@
+/******************************************************************************
+ *
+ * 
+ *
+ * Copyright (C) 2016
+ *
+ * Permission to use, copy, modify, and distribute this software and its
+ * documentation under the terms of the GNU General Public License is hereby 
+ * granted. No representations are made about the suitability of this software 
+ * for any purpose. It is provided "as is" without express or implied warranty.
+ * See the GNU General Public License for more details.
+ *
+ * Documents produced by Doxygen are derivative works derived from the
+ * input used in their production; they are not affected by this license.
+ *
+ */ /*! \page code_style Code Style for Contributors
+
+# Introduction #
+
+This document describes the code style to be used when contributing to
+the PIO library.
+
+Consistency of style allows bugs to be spotted more quickly and
+contributes to code quality.
+
+# C Code #
+
+## Backward Compatibility ##
+
+<p>We cannot remove or change a function in the public C API. We can
+add to it, carefully.
+
+## Brevity ##
+
+<p>Consider: C is a terse language.
+
+<ul>
+<li>Use the fact that NULL == 0, and 0 == false, to shorten code.
+<li>Assign and check return values in the same if statement.
+</ul>
+
+## Indentation and Spacing##
+
+<ul>
+<li>Use spaces not tabs.
+<li>4 spaces is the unit of intendation.
+<li>Indentation as defined by the "linux" style in emacs (see below).
+<li>Use spaces around most operators (=+-*/) not pointer or prefix/postfile (*++--)
+<li>Spaces after most keywords (if, for, while, etc.)
+<li>No spaces after function name.
+</ul>
+
+## Braces ##
+
+<p>Put braces on their own line, avoiding their use if possible.
```

Comment:
Yes, it's a deal-breaker. ;-)

Need them on separate lines to allow easy commenting out of if statement without having to hunt down other brace.